cpp/src/gandiva/like_holder.cc

```diff
@@ -22,10 +22,20 @@
 #include "gandiva/regex_util.h"
 
 namespace gandiva {
+RE2 LikeHolder::starts_with_regex_(R"((\w|\s|\\-)*\.\*)");
```

Comment:
maybe, we should make this more generic. Allow any characters other than . and * ?

`RE2 LikeHolder::starts_with_regex_(R"([^\.\*]*\.\*)");
`
same for ends_with & is_substr ?

Reply:
done